cro-clib/examples/rpc.c

```diff
@@ -0,0 +1,55 @@
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <stdlib.h>
+#include <string.h>
+#include <wchar.h>
+#include <assert.h>
+#include "../chain-core.h"
+#include "../chain.h"
+
+// rate: 0.0 ~ 100.0
+int32_t  progress(float rate) 
```

Comment:
The callback probably needs some context to be useful.

Reply:
some kind of struct would be fine?

Reply:
I suggest something like this:
```
int progress(int current, int total, void *data);
```
- Change rate to current and total, user has more freedom to decide how to deal with the progress information.
- Add a data pointer, whatever user passes into the api, we pass it to the callback.
- Use the return value to support aborting?

Reply:
ok